src/http_rest_api_handler.cpp

```diff
@@ -465,21 +465,22 @@ Status HttpRestApiHandler::processV3(const std::string_view uri, const HttpReque
 
         auto modelNameIt = doc.FindMember("model");
         if (modelNameIt == doc.MemberEnd()) {
-            return Status(StatusCode::JSON_INVALID, "\"model\" field is missing in JSON body");
+            return Status(StatusCode::JSON_INVALID, "model field is missing in JSON body");
         }
 
         if (!modelNameIt->value.IsString()) {
-            return Status(StatusCode::JSON_INVALID, "\"model\" field is not a string");
+            return Status(StatusCode::JSON_INVALID, "model field is not a string");
         }
 
         const std::string model_name = modelNameIt->value.GetString();
-
-        auto streamIt = doc.FindMember("stream");
-        if (streamIt != doc.MemberEnd()) {
-            if (!streamIt->value.IsBool()) {
-                return Status(StatusCode::JSON_INVALID, "\"stream\" field is not a boolean");
+        if (uri != "/v3/embeddings") {
+            auto streamIt = doc.FindMember("stream");
```

Comment:
why are we checking stream field here? it is not a part of embedding API

Reply:
this function is used also for processing completions and chat completions requests and that's why we check stream field here (but when request is meant for embeddings we skip this, check at line 476)

Reply:
will change logic to only deserialize stream when it is completion/chat completion endpoint